File: RetailChainUi/src/services/shift.service.js
// src/services/shift.service.js
// API service cho chức năng quản lý ca làm việc (Shift)

import { axiosPublic } from './api/axiosClient';

const shiftService = {
    // ==================== SHIFT CRUD ====================

    /**
     * Lấy tất cả ca làm
     */
    getAllShifts: async () => {
        const response = await axiosPublic.get('/shifts');
        return response;
    },

    /**
     * Lấy ca làm theo cửa hàng
     * @param {number} storeId
     */
    getShiftsByStore: async (storeId) => {
        const response = await axiosPublic.get(`/shifts/store/${storeId}`);
        return response;
    },

    /**
     * Tạo ca làm mới
     * @param {{ storeId: number, name: string, startTime: string, endTime: string }} data
     */
    createShift: async (data) => {
        const response = await axiosPublic.post('/shifts', data);
        return response;
    },

    // ==================== SHIFT ASSIGNMENT ====================

    /**
     * Phân công ca cho nhân viên
     * @param {{ shiftId: number, userId: number, workDate: string, createdBy: number }} data
     */
    assignShift: async (data) => {
        const response = await axiosPublic.post('/shifts/assign', data);
        return response;
    },

    /**
     * Hủy phân công ca
     * @param {number} assignmentId
     */
    cancelAssignment: async (assignmentId) => {
        const response = await axiosPublic.put(`/shifts/assign/${assignmentId}/cancel`);
        return response;
    },

    /**
     * Lấy danh sách phân công theo cửa hàng và khoảng thời gian
     * @param {number} storeId
     * @param {string} from - Format: yyyy-MM-dd
     * @param {string} to - Format: yyyy-MM-dd
     */
    getAssignments: async (storeId, from, to) => {
        const response = await axiosPublic.get('/shifts/assignments', {
            params: { storeId, from, to }
        });
        return response;
    },

    /**
     * Lấy danh sách phân công theo user
     * @param {number} userId
     */
    getAssignmentsByUser: async (userId) => {
        const response = await axiosPublic.get(`/shifts/assignments/user/${userId}`);
        return response;
    }
};

export default shiftService;

File: RetailChainService/src/main/java/com/sba301/retailmanagement/service/ShiftService.java
package com.sba301.retailmanagement.service;

import com.sba301.retailmanagement.dto.request.ShiftAssignmentRequest;
import com.sba301.retailmanagement.dto.request.ShiftRequest;
import com.sba301.retailmanagement.dto.response.ShiftAssignmentResponse;
import com.sba301.retailmanagement.dto.response.ShiftResponse;

import java.time.LocalDate;
import java.util.List;

public interface ShiftService {

    ShiftResponse createShift(ShiftRequest request);
    ShiftResponse updateShift(Long shiftId, ShiftRequest request);

    List<ShiftResponse> getAllShifts();

    List<ShiftResponse> getShiftsByStore(Long storeId);

    List<ShiftResponse> getGlobalTemplates();

    List<ShiftResponse> importTemplates(Long storeId, List<Long> templateIds);

    List<ShiftAssignmentResponse> assignShifts(ShiftAssignmentRequest request);

    ShiftAssignmentResponse cancelAssignment(Long assignmentId);

    List<ShiftAssignmentResponse> getAssignmentsByStoreAndDateRange(Long storeId, LocalDate from, LocalDate to);

    List<ShiftAssignmentResponse> getAssignmentsByUser(Long userId);
}
